src/main/java/org/openrewrite/java/spring/security5/UseNewRequestMatchers.java

```diff
@@ -0,0 +1,110 @@
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.TypeUtils;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class UseNewRequestMatchers extends Recipe {
+
+    private static final MethodMatcher ANT_MATCHERS = new MethodMatcher("org.springframework.security.config.annotation.web.AbstractRequestMatcherRegistry antMatchers(..)");
+    private static final MethodMatcher MVC_MATCHERS = new MethodMatcher("org.springframework.security.config.annotation.web.AbstractRequestMatcherRegistry mvcMatchers(..)", true);
+    private static final MethodMatcher REGEX_MATCHERS = new MethodMatcher("org.springframework.security.config.annotation.web.AbstractRequestMatcherRegistry regexMatchers(..)");
+
+
+    @NonNull
+    @Override
+    public String getDisplayName() {
+        return "Use the new `requestMatchers` methods";
+    }
+
+    @NonNull
+    @Override
+    public String getDescription() {
+        return "In Spring Security 5.8, the `antMatchers`, `mvcMatchers`, and `regexMatchers` methods were deprecated in favor of new `requestMatchers` methods. Refer to the [Spring Security docs](https://docs.spring.io/spring-security/reference/5.8/migration/servlet/config.html#use-new-requestmatchers) for more information.";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getSingleSourceApplicableTest() {
+        return Applicability.or(
+                new UsesMethod<>(ANT_MATCHERS),
+                new UsesMethod<>(MVC_MATCHERS),
+                new UsesMethod<>(REGEX_MATCHERS));
+    }
+
+    @Override
+    public JavaIsoVisitor<ExecutionContext> getVisitor() {
+        return new JavaIsoVisitor<ExecutionContext>() {
+
+            @Override
+            public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+                J.MethodInvocation mi = super.visitMethodInvocation(method, ctx);
+                if (ANT_MATCHERS.matches(mi) || MVC_MATCHERS.matches(mi) || REGEX_MATCHERS.matches(mi)) {
+                    mi = maybeChangeMethodInvocation(mi);
+                }
+                return mi;
+            }
+        };
+
+    }
+
+    private J.MethodInvocation maybeChangeMethodInvocation(J.MethodInvocation mi) {
+        JavaType.Method requestMatchersMethod = findRequestMatchersMethodWithMatchingParameterTypes(mi);
+        if (requestMatchersMethod != null) {
+            return mi
+                    .withMethodType(requestMatchersMethod)
+                    .withName(mi.getName().withSimpleName("requestMatchers"));
+        } else {
+            return mi;
+        }
+    }
+
+    @Nullable
+    private JavaType.Method findRequestMatchersMethodWithMatchingParameterTypes(J.MethodInvocation mi) {
+        JavaType.Method methodType = requireNonNull(mi.getMethodType(), "methodType");
```

Comment:
I think the method matchers already guard against a null `methodType` here; and even if we do get a null methodType, this recipe specifically is for Spring Boot 3, which requires Java 17, which will give us helpful null pointer exceptions further down. I may be wrong of course, but given the above I don't think there's much value in the explicit validation.
```suggestion
        JavaType.Method methodType = mi.getMethodType();
```

Reply:
It really is a matter of preference I guess. If you were to be picky the recipe is not specifically for Boot 3, so JDK17 is not a given.

But realistically: In 98% it _will_ be a Boot 3 migration